### ISSUE TITLE
unexpected End of file

### ISSUE THREAD

**Issue by shamen123** [savonet/liquidsoap#2065]
**Describe the bug**
When a source connects via input.harbor I mux the audio with a video mp4 file and then throw that into an ffmpeg encoder and send it out over RTMP. Sources can connect and disconnect but seeminly at random, the ffmpeg decoder of the video will behave differently. 

eg, when a source connects - my file somevideofile.mp4 (approx 60 minutes video) will play on loop. This shows in the logs like so 

2021/11/19 11:37:00 [decoder:4] Decoding "somevideofile.mp4" ended: Ffmpeg_decoder.End_of_file.
2021/11/19 11:37:00 [decoder:4] Raised at file "decoder/ffmpeg_decoder.ml", line 691, characters 12-29
2021/11/19 11:37:00 [decoder:4] Called from file "decoder/decoder.ml", line 500, characters 10-31
2021/11/19 11:37:00 [decoder:4] 
2021/11/19 11:37:01 [somevideofile.mp4:4] Finished with "somevideofile.mp4".
2021/11/19 11:37:01 [somevideofile.mp4:3] Prepared "somevideofile.mp4" (RID 0).
2021/11/19 11:37:01 [somevideofile.mp4:5] Got metadata at position 0: calling handlers...
2021/11/19 11:37:01 [mux_video_0:5] Got metadata at position 0: calling handlers...
2021/11/19 12:37:35 [decoder:4] Decoding "somevideofile.mp4" ended: Ffmpeg_decoder.End_of_file.
2021/11/19 12:37:35 [decoder:4] Raised at file "decoder/ffmpeg_decoder.ml", line 691, characters 12-29
2021/11/19 12:37:35 [decoder:4] Called from file "decoder/decoder.ml", line 500, characters 10-31
2021/11/19 12:37:35 [decoder:4] 
2021/11/19 12:37:35 [mux_video_0:5] Got metadata at position 0: calling handlers...
2021/11/19 12:38:12 [input.ffmpeg_0:4] Feeding failed: Ffmpeg_decoder.End_of_file
2021/11/19 12:38:12 [input.ffmpeg_0:4] Raised at file "decoder/ffmpeg_decoder.ml", line 691, characters 12-29
2021/11/19 12:38:12 [input.ffmpeg_0:4] Called from file "io/ffmpeg_io.ml", line 174, characters 10-24
2021/11/19 12:38:12 [input.ffmpeg_0:4] 

the file "somevideofile.mp4" exists as it has been played at 11:37 and reloaded at 12:37:35 .. The random part is what happens at 12:38:12 - this can happen quickly or hours into the loop of the video. When that EOF on input.ffmpeg happens, it no longer matters if a DJ connects to input.harbor , the rtmp stream stops. 

I get that it is the failure in the input mp4, but i cant see a reason for that. the file is good and exists. there is no OOM killer happening. liquidsoap / ffmpeg read the file just fine in the previous hours. How can I identify why there is this EOF on the mp4 input, and do you have any config recommendations that I could use as a workaround to auto heal from it?

**To Reproduce**

code

```
set("log.level", 7)
set("harbor.bind_addr","0.0.0.0")
set("harbor.verbose",true)
video = single("somevideofile.mp4")
source = input.harbor("/",icy=true,port=8000,password="somepassword")
emergency = single("emergency.mp3")
sound = fallback(track_sensitive=false,[source,emergency])
stream = mux_video(video=video,sound)
enc = %ffmpeg(format="flv",%audio(codec="aac",samplerate=44100,b="128k"),%video.copy)
url = "rtmp://rtmp.server/mount/key"
output.url(url=url, enc, stream)
```

**Expected behavior**
the mp4 file will mux with the harbor input and output to RTMP server

**Version details**
 - OS: ubuntu 18.04 LTS
 - Version 2.0.0

**Install method**
OPAM install

**Comment by toots** [comment]
Thanks for this report. This is a very interesting use-case. I'm gonna do my best to look at it very soon.

**Comment by shamen123** [comment]
Hi @toots  thanks for checking. when the problem occurs, it drops back to emergency and stays on emergency even if a new source connects to input.harbor .. so my RTMp server gets emergency.mp3 on a loop, no video and the source can reconnect over and over nothing happens.

**Comment by shamen123** [comment]
hi @toots I think the problem report logs are skewed as I found that my debug logs had both instances outputting to same log file. I have now separated them. I run two instances 
Instance 1: harbor audio input and mp4 file video mux that sends RTMP output to server - as per above config file
Instance 2: RTMP client (input.ffmpeg) stream receiving from server and outputting to shoutcast. 

It is instance 2 that gets the ffmpeg EOF. But it seems to be triggered by instance 1 still taking harbor input but not outputting it to the RTMP server. What it looks like is happening is if the source 'falls back' to emergency.mp3 it no longer mux'es the video and outputs to ffmpeg

**Comment by shamen123** [comment]
hi @toots  I think the problem report logs are skewed as I found that my debug logs had both instances outputting to same log file. I have now separated them. I run two instances
Instance 1: harbor audio input and mp4 file video mux that sends RTMP output to server - as per above config file
Instance 2: RTMP client (input.ffmpeg) stream receiving from server and outputting to shoutcast.

It is instance 2 that gets the ffmpeg EOF. At this stage im going to concentrate my investigation on the RTMP server. I fear the liquidsoap behaviour is a symptom, not the cause. 

ill get more logs and come back to this report with more data or close it out if its server side.
